src/OpenTelemetry/Metrics/MetricStreamIdentity.cs

```diff
@@ -37,9 +37,30 @@ public MetricStreamIdentity(Instrument instrument, MetricStreamConfiguration met
             this.HistogramBucketBounds = (metricStreamConfiguration as ExplicitBucketHistogramConfiguration)?.CopiedBoundaries;
             this.HistogramRecordMinMax = (metricStreamConfiguration as HistogramConfiguration)?.RecordMinMax ?? true;
 
+#if NETSTANDARD2_1 || NET6_0_OR_GREATER
+            HashCode hashCode = default;
+            hashCode.Add(this.InstrumentType.GetHashCode());
+            hashCode.Add(this.MeterName.GetHashCode());
+            hashCode.Add(this.MeterVersion.GetHashCode());
+            hashCode.Add(this.InstrumentName.GetHashCode());
+            hashCode.Add(this.HistogramRecordMinMax.GetHashCode());
+            hashCode.Add(this.Unit.GetHashCode());
+            hashCode.Add(this.Description.GetHashCode());
+            hashCode.Add(this.ViewId?.GetHashCode() ?? 0);
+            hashCode.Add(this.TagKeys != null ? StringArrayComparer.GetHashCode(this.TagKeys) : 0);
```

Comment:
There is an overload which accepts `IEqualityComparer<T>` you might be able to use here:

```csharp
hashCode.Add(this.TagKeys, StringArrayComparer);
```

Reply:
Updated.